service/resource/endpoint/current.go

```diff
@@ -0,0 +1,39 @@
+package endpoint
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework/context/canceledcontext"
+)
+
+const (
+	ipAnnotation      = "endpoint.kvm.giantswarm.io/ip"
+	serviceAnnotation = "endpoint.kvm.giantswarm.io/service"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	pod, err := toPod(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	ipAnnotationValue, serviceAnnotationValue, err := getAnnotations(*pod, ipAnnotation, serviceAnnotation)
+	if IsMissingAnnotationError(err) {
+		canceledcontext.SetCanceled(ctx)
+		if canceledcontext.IsCanceled(ctx) {
+			r.logger.Log("pod", pod.GetName(), "debug", "canceling reconciliation for pod")
+			return nil, nil
+		}
+	} else if err != nil {
+		return nil, microerror.Maskf(err, "an error occurred while fetching the annotations of the pod")
```

Comment:
Adding annotations for errors - I really appreciate that <3

Hint from me. When you return error expect it to be logged as `error: cause: annotation`.
In this case `error: missing annotation: an error occurred while fetching the annotations of the pod`
It feels like the information is repeated
I think it'd be better to log additional information which makes debugging easy. E.g. `error: missing annotation: pod=POD_NAME`